internal/pkg/manifest/workload.go

```diff
@@ -397,6 +406,7 @@ type DockerBuildArgs struct {
 	Args       map[string]string `yaml:"args,omitempty"`
 	Target     *string           `yaml:"target,omitempty"`
 	CacheFrom  []string          `yaml:"cache_from,omitempty"`
+	Engine     *string           `yaml:"engine,omitempty"`
```

Comment:
Since essentially `nil` and `""` are the same case (we'll set to default value "docker"), maybe we can use `string` instead of `*string`